src/swoop/api/routers/processes.py

```diff
@@ -138,17 +149,64 @@ async def get_process_definition(process_id: str = Path(..., alias="processID"))
 
 @router.post(
     "/{process_id}/execution",
-    response_model=InlineResponse200,
+    response_model=StatusInfo,
     responses={
         "201": {"model": StatusInfo},
         "404": {"model": APIException},
-        "500": {"model": APIException},
+        "422": {"model": APIException},
     },
 )
-def execute_process(
-    process_id: str = Path(..., alias="processID"), body: Execute = ...
-) -> InlineResponse200 | StatusInfo | APIException:
+async def execute_process(
+    process_id: str, request: Request, body: Execute
+) -> StatusInfo | APIException:
     """
     execute a process.
     """
-    pass
+
+    workflows = (
+        request.app.state.workflows.get("workflows")
+        if request.app.state.workflows.get("workflows")
+        else {}
+    )
+
+    if process_id and len(workflows.keys()) > 0:
+        if process_id in workflows:
+            workflow = workflows[process_id]
+        else:
+            raise HTTPException(status_code=404, detail="Process not found")
+
+    # Generate a UUID
+    action_uuid = uuid.uuid4()
+
+    # Write to object storage
+
+    request.app.state.io.put_object(
+        object_name=f"executions/{action_uuid}/input.json",
+        object_content=json.dumps(body.dict()["inputs"]),
+    )
+
+    async with request.app.state.readpool.acquire() as conn:
+        # Insert a dummy row into payload_cache table
+        # To be removed later
+        pl_uuid = "debeb36c-e09e-41c3-bdc5-596287fe724a"
+
+        q, p = render(
+            f"""
+                INSERT INTO swoop.payload_cache(payload_uuid,
+                    workflow_version, workflow_name)
+                VALUES ('{pl_uuid}',2,'mirror');
+            """,
+        )
+        await conn.fetchrow(q, *p)
+
+        q, p = render(
+            f"""
+                INSERT INTO swoop.action(action_uuid, action_type,
+                    action_name, handler_name, parent_uuid, payload_uuid)
+                VALUES ('{action_uuid}','workflow','{workflow.name}',
+                '{workflow.handler}','{uuid.uuid4()}','{pl_uuid}');
+            """,
```

Comment:
If you find yourself using string templating for sql queries, _stop immediately_.

Please look at the buildpg documentation and other API queries for examples of how to write queries that are not vulnerable to sql injection.